### PR TITLE
add type definitions

### PR DESCRIPTION
**Pull request by isaacs**
- [ ] port all tests to typescript, so we know the types are actually correct

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
+/// <reference types="node" />
+import { EventEmitter } from 'events'
+
+type Encoding = BufferEncoding | 'buffer' | null
+
+interface Writable extends EventEmitter {
```
Comment (benjamingr) [inline index.d.ts:6]: Q: Any reason this doesn't reference the Stream type instead since Minipass extends the Stream type?

Comment (isaacs) [reply index.d.ts:6]: I was thinking that, but the definitely-typed version extends EventEmitter for some reason?

Comment (isaacs) [reply index.d.ts:7]: Yeah, it seems to work fine extending from Stream (which is just an EventEmitter + `pipe(dest, options?)` after all), so I went that direction with it.

**Comment by jimlindeman** [comment]
Frequently seeing this tsc error:
```
node_modules/minipass/index.d.ts(79,18): error TS1005: ',' expected.
node_modules/minipass/index.d.ts(80,17): error TS1005: ',' expected.
```
because of this change.   This is with "typescript": "^3.9.9" in our package.json

**Comment by isaacs** [comment]
@jimlindeman What version of minipass do you have?

Can you upload your `node_modules/minipass/index.d.ts` file to https://gist.github.com and share the link, or attach it to a comment here?

**Comment by isaacs** [comment]
I'm using Typescript version 4.7.7, so it's likely that the types are using something new-ish, I guess?

**Comment by isaacs** [comment]
Should be fixed by 0875015.  TypeScript 3 didn't have support for named conditional arguments in the same way that TS 4 supports, I guess.

**Pull request merged**